.github/workflows/tiledb-cloud-py.yaml

```diff
@@ -107,10 +120,27 @@ jobs:
             pytest -sv \
             --last-failed \
             --tb short \
-            --color yes
+            --color yes \
+            --splits ${{ env.PYTEST_SPLIT_GROUPS }} --group ${{ matrix.pytest-split-group }} \
+            --store-durations --splitting-algorithm least_duration --durations-path=.test_durations.${{ matrix.pytest-split-group }} --clean-durations
         env:
           TILEDB_REST_TOKEN: ${{ secrets.TILEDB_CLOUD_HELPER_VAR }}
 
+  save-test-durations-to-cache:
+    needs: run-tests
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - name: "Combine test duration cache files in .test_durations"
```

Comment:
- Do we need to update this in every run?
- Is the action here going to commit back to `.test_durations` on this branch, or is that just a cache key?

Reply:
My understanding here:

- this saves the timings from the latest run to the cache, and does not modify what is in the repository
- we do this so pytest has an idea of how to partition tests so that the N shards run in about the same length of time

It seems reasonable from a "how" perspective, but having a better explanation in a comment would be useful.

It also seems like maybe we should include the matrix variables in the cache key, so that all the different matrix entries aren’t continually overwriting each other’s timing values.

Reply:
@ihnorton @thetorpedodog  I will update the commit with a detailed explanation when it is ready for a final review.

Reply:
@ihnorton @thetorpedodog I have updated the PR text to explain the process.